### ISSUE TITLE
xlearn always be killed

### ISSUE THREAD

**Issue by rookie77** [aksnzhy/xlearn#174]
hi,
my train data  size is  around 27G, and my validation data is 15G, after the transformation the train .bin is 39G and val.bin is 21G. but the job always be killed, is that normal? can you help me to fix this?
my device info is 8cores 64g+200G 。centerOs6

**Comment by rookie77** [comment]
feature num is around 129w~~

**Comment by aksnzhy** [comment]
@rookie77 Hi, can you please share your code related with the xLearn part? I will try to help you figure out the problem.

**Comment by rookie77** [comment]
> @rookie77 Hi, can you please share your code related with the xLearn part? I will try to help you figure out the problem.
thanks for the reply~ here is my code, it's very simple 
######
 lr_model = xl.create_linear()
 #    lr_model.setOnDisk()
 lr_model.setTXTModel(args.outTxtModel)
 lr_model.setTrain(args.trainData)
print("==========")
 lr_model.setValidate(args.testData)

 lr_model.setTest(args.testData)
 
  param = {'task': 'binary', 'lr': args.lr, 'lambda': args.Lambda, 'metric': args.metric, 'epoch':10}


lr_model.fit(param, "./model.out")
 lr_model.setSigmoid()
 lr_model.predict("./model.out", "./output.txt")

**Comment by rookie77** [comment]
lr= 0.002, Lambda=0.02 metric ='auc' 
one more question: will xlearn shuffle the train data automaticly?

**Comment by aksnzhy** [comment]
@rookie77 Hi, I looks like that the memory cost is beyond the capability of the local machine. 

(39 + 21 + some other cost) > 64g memory (maybe the real memory can be used in xLearn is less than 64g).

You can try this:  

Divide fit() and predict() into separated python process.

or

Use on-disk training for this task.
